src/main/java/de/tum/in/www1/artemis/service/connectors/localci/buildagent/BuildJobExecutionService.java

```diff
@@ -416,6 +419,34 @@ else if (!"skipped".equals(xmlStreamReader.getLocalName())) {
         }
     }
 
+    private String readTestMessageFromLegacyReport(XMLStreamReader xmlStreamReader) throws XMLStreamException {
+        // JUnit legacy report format:
+        // The old report format does not use the message attribute, but instead has the error message as a child element
```

Comment:
I'm not sure what makes this the "legacy" format, is there any indication that tools (JUnit itself or third-party) don't produce this output anymore?

Is there any documentation or reference implementation that Artemis tries to follow? E.g. https://github.com/windyroad/JUnit-Schema/blob/master/JUnit.xsd ?

Reply:
I would be interested in some kind of reference/link in the comment too. In order for other developers to be able to look it up.

Reply:
I name it legacy since I remembered reading about a report format change in the JUnit change logs at some point.

Looking it up now, I was thinking of Release 5.9.0: https://junit.org/junit5/docs/5.9.0/release-notes/

> XML reports in new [Open Test Reporting](https://github.com/ota4j-team/open-test-reporting) format

However, this format is not active by default and we actually need to add support for this in the future.